src/render/tutorials/api-gateway-security/2_0.html.md

```diff
@@ -0,0 +1,954 @@
+---
+title: API gateway Security
+author: Mateusz512
+keywords: tutorial
+date: 2019-10-28
+layout: tutorial
+knotxVersions:
+  - 2.0.0
+---
+## Overview
+
+In this tutorial, we will take a look on how to secure an API gateway.
+
+This tutorial is based on the code we implemented in [Getting Started with Docker tutorial](/tutorials/getting-started-with-docker/). If you didn't complete the tutorial, we highly encourage you to do so.
+
+### Prerequisites
+
+- JDK 8
+- Docker (not required, you can use Stack distribution as well)
+- [Postman](https://www.getpostman.com/) (it will simplify things, but `curl` is fine as well) 
+- Modern web browser
+- Google account
+
+To make things simple, we'll start from the code already implemented in the tutorial mentioned before, which is available [here](https://github.com/Knotx/knotx-example-project/tree/master/distribution/getting-started).
+
+Let's quickly recap on the things we'll reuse here.
+
+We already implemented the `HelloWorldHandlerFactory` which simply prints the configured message to the HTTP response:
+
+```java
+// ...
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    String message = config.getString("message", "Some default message");
+    JsonObject jsonObject = new JsonObject()
+        .put("message", message);
+    return event -> event.response().end(jsonObject.toString());
+  }
+
+// ...
+```
+
+In this tutorial we'll make sure that this handler is accessible only for authenticated users.
+
+## Basic auth
+
+Let's start with the most basic way of authenticating our requests - [Basic auth](https://en.wikipedia.org/wiki/Basic_access_authentication).
+
+### Design & Setup
+
+As always, we'll use the [Design First approach](https://swagger.io/blog/api-design/design-first-or-code-first-api-development/), therefore we'll modify our [Open API 3](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md) configuration file. Let's add two new endpoints in the `paths:` section and a new `securityScheme`  in the [`components`](https://swagger.io/docs/specification/components/) section:
+ 
+`knotx/conf/openapi.yaml`
+```yaml
+// ...
+
+paths:  
+  /api/public:  // a public endpoint with no security, renamed from /api/hello
+    get:
+      operationId: hello-world-operation  // the same operation as in /api/hello
+      responses:
+        default:
+          description: Public Hello World API
+  /api/secure/basic:  // endpoint with basic authentication authentication
+    get:
+      operationId: hello-world-operation-basic
+      security:
+        - helloBasicAuth: []  // here we specify which security scheme we'll be using (see below)
+      responses:
+        default:
```

Comment:
Can we specify here responses? There should be `200` and `401` responses. See https://swagger.io/docs/specification/authentication/basic-authentication/.

Reply:
It is explained in so you can link this tutorial: https://knotx.io/tutorials/openapi-and-swagger-ui/2_0/